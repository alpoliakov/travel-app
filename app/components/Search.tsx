import React from 'react';
import { useIntl } from 'react-intl';

export default function Search() {
  const { formatMessage: f } = useIntl();

  return (
    <div className="flex items-center justify-center">
      <form>
        <div className="relative text-gray-600 focus-within:text-gray-400">
          <span className="absolute inset-y-0 left-0 flex items-center pl-2">
            <button type="submit" className="p-1 focus:outline-none focus:shadow-outline">
              <svg
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                className="w-6 h-6">
                <path d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" />
              </svg>
            </button>
          </span>
          <input
            type="search"
            name="q"
            className="py-2 text-sm text-white bg-gray-800 rounded-md pl-10 pr-2 focus:outline-none focus:bg-white focus:text-gray-900"
            placeholder={`${f({ id: 'search' })}...`}
            autoComplete="off"
          />
        </div>
      </form>
    </div>
  );
}
