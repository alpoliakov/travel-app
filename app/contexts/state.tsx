import React, { createContext, useContext, useState } from 'react';

const AppContext = createContext(null);

export function AppWrapper({ children }) {
  const [text, setText] = useState('');

  const sharedState = {
    text: text,
    setText: (value) => {
      setText(value);
    },
  };

  return <AppContext.Provider value={sharedState}>{children}</AppContext.Provider>;
}

export function useAppContext() {
  return useContext(AppContext);
}
