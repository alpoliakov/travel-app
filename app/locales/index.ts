export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    return: 'Return ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    capital: 'Capital',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
    ratingQuestion: 'How do you rate this place?',
    opinion: 'Your opinion matters to us!',
    rateButton: 'Rate now',
    later: 'Maybe later',
    reminder: 'You need to login or register!',
    feedbackGive: 'give feedback',
    feedbackSee: 'see reviews',
    usd: 'USD',
    eur: 'EUR',
    rub: 'RUB',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    return: 'Вернуться ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    capital: 'Столица',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
    ratingQuestion: 'Как вы оцениваете это место?',
    opinion: 'Ваше мнение важно для нас!',
    rateButton: 'Оценить сейчас',
    later: 'Может быть, позже',
    reminder: 'Вам необходимо войти или зарегистрироваться!',
    feedbackGive: 'оставить отзыв',
    feedbackSee: 'посмотреть отзывы',
    usd: 'Доллар',
    eur: 'Евро',
    rub: 'Рубль',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    return: 'Повернутись ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    capital: 'Столиця',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
    ratingQuestion: 'Як ви оцінюєте це місце?',
    opinion: 'Ваша думка для нас важлива!',
    rateButton: 'Оцініть зараз',
    later: 'Можливо пізніше',
    reminder: 'Вам необхідно увійти або зареєструватися!',
    feedbackGive: 'залишити відгук',
    feedbackSee: 'подивитися відгуки',
    usd: 'Долар',
    eur: 'Євро',
    rub: 'Рубль',
  },
};
