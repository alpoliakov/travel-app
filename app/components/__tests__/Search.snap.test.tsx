import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { AppWrapper } from '../../contexts/state';
import { messages } from '../../locales';
import Search from '../Search';

it('Search render correctly', () => {
  const tree = renderer
    .create(
      <AppWrapper>
        <IntlProvider locale="ru" messages={messages['ru']}>
          <Search />
        </IntlProvider>
      </AppWrapper>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
