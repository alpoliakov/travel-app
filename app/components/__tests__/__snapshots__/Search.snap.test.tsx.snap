// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render correctly 1`] = `
<div
  className="flex items-center justify-center"
>
  <form>
    <div
      className="relative text-gray-600 focus-within:text-gray-400"
    >
      <span
        className="absolute inset-y-0 left-0 flex items-center pl-2"
      >
        <button
          aria-label="button"
          className="p-1 focus:outline-none focus:shadow-outline"
          type="button"
        >
          <svg
            className="w-6 h-6"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            />
          </svg>
        </button>
      </span>
      <input
        autoComplete="off"
        className="py-2 text-sm text-white bg-gray-800 rounded-md pl-10 pr-2 focus:outline-none focus:bg-white focus:text-gray-900"
        name="q"
        onChange={[Function]}
        placeholder="Поиск..."
        type="search"
        value=""
      />
    </div>
  </form>
</div>
`;
